backend/src/services/course-modules/course-module.service.ts

```diff
@@ -0,0 +1,65 @@
+import { ExceptionMessage, HttpCode } from '~/common/enums/enums';
+import {
+  CourseModuleCreateArgumentsDto,
+  CourseModuleGetResponseDto,
+} from '~/common/types/types';
+import { courseModule as moduleRep } from '~/data/repositories/repositories';
+import { CoursesModulesError } from '~/exceptions/exceptions';
+import { udemyCourseModule as udemyServ } from '~/services/services';
+
+type Constructor = {
+  moduleRepository: typeof moduleRep;
+  udemyService: typeof udemyServ;
+};
+
+class CourseModule {
+  #moduleRepository: typeof moduleRep;
+
+  #udemyService: typeof udemyServ;
+
+  public constructor({ moduleRepository, udemyService }: Constructor) {
+    this.#moduleRepository = moduleRepository;
+    this.#udemyService = udemyService;
+  }
+
+  public create(
+    moduleRequestDto: CourseModuleCreateArgumentsDto,
+  ): Promise<CourseModuleGetResponseDto> {
+    const { title, description, moduleIndex, courseId } = moduleRequestDto;
+
+    return this.#moduleRepository.create({
+      title,
+      description,
+      moduleIndex,
+      courseId,
+    });
+  }
+
+  public async createModulesByCourseId(
+    serviceCourseId: number,
+    dbCourseId: number,
+  ): Promise<void> {
+    const courseData = await this.#udemyService.getModulesByCourseId(
+      serviceCourseId,
+    );
+
+    if (!courseData.length) {
+      throw new CoursesModulesError({
+        status: HttpCode.BAD_GATEWAY,
```

Comment:
```suggestion
        status: HttpCode.BAD_GATEWAY,
```
our server should throw only 4** errors

Reply:
Roman added that to checklist for that ticket, that if modules are empty, then we throw 502 or 503 error, or 4** enough for now?